src/test/scala/strawman/collection/test/GenericTest.scala

```diff
@@ -0,0 +1,59 @@
+package strawman
+package collection
+package test
+
+import mutable.Builder
+
+import scala.util.Try
+import scala.{Any, Int, None, Nothing, Option, Ordering, Some, Unit}
+import scala.Predef.{augmentString, implicitly, wrapRefArray, assert}
+import java.lang.String
+
+import org.junit.Test
+
+trait Parse[A] {
+  def parse(s: String): Option[A]
+}
+
+object Parse {
+
+  implicit def parseInt: Parse[Int] = (s: String) => Try(s.toInt).toOption
+
+  implicit def parseTuple[A, B](implicit
+                                parseA: Parse[A],
+                                parseB: Parse[B]
+                               ): Parse[(A, B)] = { (s: String) =>
+    val parts = s.split("-")
+    (parseA.parse(parts(0)), parseB.parse(parts(1))) match {
+      case (Some(a), Some(b)) => Some((a, b))
+      case _ => None
+    }
+  }
+
+  def parseCollection[A, C](bf: BuildFrom[Any, A, C])
+                                    (implicit parseA: Parse[A]): Parse[C] = { (s: String) =>
+    val parts = s.split("\\|")
+    parts.foldLeft[Option[Builder[A, C]]](Some(Builder.from(bf, null))) { (maybeBuilder, s) =>
```

Comment:
`FromSpecificIterable` should be sufficient. AFAIR I ran into some problem with the implicits with that.